external/storm-druid/src/main/java/org/apache/storm/druid/bolt/DruidBeamBolt.java

```diff
@@ -0,0 +1,115 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.druid.bolt;
+
+import com.metamx.tranquility.tranquilizer.MessageDroppedException;
+import com.metamx.tranquility.tranquilizer.Tranquilizer;
+import com.twitter.util.Future;
+import com.twitter.util.FutureEventListener;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.topology.base.BaseRichBolt;
+import org.apache.storm.tuple.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Basic bolt implementation for storing data to Druid datastore.
+ * <p/>
+ * This implementation uses Druid's Tranquility library (https://github.com/druid-io/tranquility)
+ * to send to druid store.
+ * Some of the concepts are borrowed from Tranquility storm connector implementation.
+ * (https://github.com/druid-io/tranquility/blob/master/docs/storm.md)
+ *
+ * This Bolt expects to receive tuples in which the zeroth element is your event type.
+ * <p/>
+ *
+ */
+public class DruidBeamBolt<E> extends BaseRichBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(DruidBeamBolt.class);
+
+    private volatile  OutputCollector collector;
+    private DruidBeamFactory<E> beamFactory = null;
+    private int batchSize;
+    private Tranquilizer<E> tranquilizer = null;
+
+    public DruidBeamBolt(DruidBeamFactory<E> beamFactory) {
+        this(beamFactory, 2000);
+    }
+
+    public DruidBeamBolt(DruidBeamFactory<E> beamFactory, int batchSize) {
+        this.beamFactory = beamFactory;
+        this.batchSize = batchSize;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        this.collector = collector;
+        tranquilizer = Tranquilizer.create(
+                beamFactory.makeBeam(stormConf, context),
+                batchSize,
+                Tranquilizer.DefaultMaxPendingBatches(),
+                Tranquilizer.DefaultLingerMillis());
+        this.tranquilizer.start();
+
+    }
+
+    @Override
+    public void execute(final Tuple tuple) {
+      Future future = tranquilizer.send((E)tuple.getValue(0));
+      future.addEventListener(new FutureEventListener() {
+          @Override
+          public void onFailure(Throwable cause) {
+              if(cause instanceof MessageDroppedException) {
```

Comment:
> MessageDroppedException: Exception indicating that a message was dropped "on purpose" by the beam. This is not a recoverable exception and so the message must be discarded.

These tuples can be acked and send all these discarded tuples to a given stream so that user can know which tuples are discarded and take an appropriate action.